StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentMethodType.swift

```diff
@@ -165,12 +165,29 @@ extension PaymentSheet {
                 // External Payment Methods
                 + elementsSession.externalPaymentMethods.map { PaymentMethodType.external($0) }
 
-            if
-                elementsSession.orderedPaymentMethodTypes.contains(.link),
-                !elementsSession.orderedPaymentMethodTypes.contains(.USBankAccount),
-                !intent.isDeferredIntent,
+            // We should manually add Instant Debits as a payment method when:
+            // - Link is an available payment method.
+            // - US Bank Account is *not* an available payment method.
+            // - Not a deferred intent flow.
+            // - Link Funding Sources contains Bank Account.
+            var eligibleForInstantDebits: Bool {
+                elementsSession.orderedPaymentMethodTypes.contains(.link) &&
+                !elementsSession.orderedPaymentMethodTypes.contains(.USBankAccount) &&
+                !intent.isDeferredIntent &&
                 elementsSession.linkFundingSources?.contains(.bankAccount) == true
-            {
+            }
```

Comment:
does this need to be a `var` instead of a `let`?

and out of curiosity/learning how does this compile?

I would have expected it to be something like:

```
            let eligibleForInstantDebits: Bool = {
                return elementsSession.orderedPaymentMethodTypes.contains(.link) &&
                !elementsSession.orderedPaymentMethodTypes.contains(.USBankAccount) &&
                !intent.isDeferredIntent &&
                elementsSession.linkFundingSources?.contains(.bankAccount) == true
            }()
```

I get the inferred `return` as they are no longer necessary.

But what are all the other pieces? 

it almost looks like an in-line computed property but the declaration is inside of a function 🤔 so I am not immediately sure the step-by-step execution for it to compile into a boolean